### PR TITLE
SUS-3101: Remove JOIN to wikicities_cX.user in ContribsPager

### PR DESCRIPTION
**Pull request by mszabo-wikia**
https://wikia-inc.atlassian.net/browse/SUS-3101

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 24 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 6 minor
* ![INFO][INFO] 11 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialContributions.php#L38](https://github.com/Wikia/app/blob/97c56bb1c2eb3104e15d1816136f14ec27546241/includes/specials/SpecialContributions.php#L38): The Cyclomatic Complexity of this function "execute" is 22 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [SpecialContributions.php#L38](https://github.com/Wikia/app/blob/97c56bb1c2eb3104e15d1816136f14ec27546241/includes/specials/SpecialContributions.php#L38): Reduce the number of returns of this function 5, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [SpecialContributions.php#L38](https://github.com/Wikia/app/blob/97c56bb1c2eb3104e15d1816136f14ec27546241/includes/specials/SpecialContributions.php#L38): This function "execute" has 152 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [SpecialContributions.php#L76](https://github.com/Wikia/app/blob/97c56bb1c2eb3104e15d1816136f14ec27546241/includes/specials/SpecialContributions.php#L76): Change the output format of this i18n string away from plain(). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3APlainFormatWfMessage)
1. ![MAJOR][MAJOR] [SpecialContributions.php#L163](https://github.com/Wikia/app/blob/97c56bb1c2eb3104e15d1816136f14ec27546241/includes/specials/SpecialContributions.php#L163): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [SpecialContributions.php#L195](https://github.com/Wikia/app/blob/97c56bb1c2eb3104e15d1816136f14ec27546241/includes/specials/SpecialContributions.php#L195): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. ![MAJOR][MAJOR] [SpecialContributions.php#L332](https://github.com/Wikia/app/blob/97c56bb1c2eb3104e15d1816136f14ec27546241/includes/specials/SpecialContributions.php#L332): This function "getForm" has 164 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MINOR][MINOR] [SpecialContributions.php#L492](https://github.com/Wikia/app/blob/97c56bb1c2eb3104e15d1816136f14ec27546241/includes/specials/SpecialContributions.php#L492): Immediately return this expression instead of assigning it to the temporary variable "$form". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. ![MINOR][MINOR] [SpecialContributions.php#L519](https://github.com/Wikia/app/blob/97c56bb1c2eb3104e15d1816136f14ec27546241/includes/specials/SpecialContributions.php#L519): Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)
1. ![MINOR][MINOR] [SpecialContributions.php#L520](https://github.com/Wikia/app/blob/97c56bb1c2eb3104e15d1816136f14ec27546241/includes/specials/SpecialContributions.php#L520): Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by macbre — APPROVED** [review]

File: includes/specials/SpecialContributions.php [hunk 1]
```diff
@@ -552,14 +552,11 @@ function getQueryInfo() {
...
 		# Don't include orphaned revisions
 		$join_cond['page'] = Revision::pageJoinCond();
-		# Get the current user name for accounts
-		$join_cond['user'] = Revision::userJoinCond();
 
 		$queryInfo = array(
 			'tables'     => $tables,
 			'fields'     => array_merge(
 				Revision::selectFields(),
-				Revision::selectUserFields(),
```
Comment (ghost) [inline includes/specials/SpecialContributions.php:562]: Is this one still needed anywhere?

**Review by ghost — APPROVED** [review]
Just one question about one method - should we keep it or remove as well?

File: includes/specials/SpecialContributions.php (hunk 1 above)
Comment (mszabo-wikia) [reply includes/specials/SpecialContributions.php:562]: It's still used in few other places, let's remove after those tickets are handled.

**Pull request merged**